packages/@lwc/engine-dom/src/renderer.ts

```diff
@@ -222,6 +222,10 @@ export const renderer: Renderer<Node, Element> = {
         return element.getElementsByClassName(names);
     },
 
+    getNearestHost(node: Node): Node | undefined {
+        return (node.getRootNode() as ShadowRoot).host;
```

Comment:
It would be great if I could just call the actual native `getRootNode()` here, but I don't think I can because we would need some way for `@lwc/synthetic-shadow` to pass the unpatched function to `@lwc/engine-dom`, and that doesn't seem possible.